File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> num_index;
        vector<int> ans(2);
        for(int i=0; i<nums.size(); i++){
            
            if(num_index.find(target - nums[i]) != num_index.end()){
                ans[0] = num_index[target - nums[i]];
                ans[1] = i;
                break;
            }
            else{
                num_index[nums[i]] = i;
            }
        }
        return ans;
    }
};

File: 0001-two-sum/0001-two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    const num_index = new Map<number, number>();
    const ans :number[] = [0,0];
    
    for(let i=0; i<nums.length; i++){
        if(num_index.has(target - nums[i])){
            ans[0] = num_index.get(target - nums[i]);
            ans[1] = i;
            break;
        }
        num_index.set(nums[i], i);  
        
    }
    return ans;
};